src/server/rpc/procedures/geolocation/geolocation.js

```diff
@@ -0,0 +1,163 @@
+        }
+    });
+    arr2.shift(); // remove the first item which is always empty
+
+    arr2.forEach(q=>{
+        res = res[q];
+    });
+    if (typeof(res) === 'object') {
+        res = JSON.stringify(res);
+    }
+    return res;
+}
+
+// reverse geocoding helper, doesn't return a promise. Hanldes sending of response.
+let reverseGeocode = (lat, lon, response, query)=>{
+    cache.wrap(lat + ', ' + lon, cacheCallback => {
+        trace('Geocoding (not cached)', lat, lon);
+        geocoder.reverse({lat, lon})
+        .then(function(res) {
+            // only intereseted in the first match
+            res = queryJson(res[0], query);
+            // send the response to user
+            return cacheCallback(null, res);
+        })
+        .catch(err => {
+            error('Error in reverse geocoding', err);
+            // showError('Failed to reverse geocode',response);
+            return cacheCallback('Error in reverse geocoding', null);
+        });
+    }, (err, results) => {
+        if(results){
+            response.send(results);
+        }else {
+            showError(err, response);
+        }
+    });
+};
+
+
+// geocode an address and send back the details
+GeoLocationRPC.geolocate = function (address) {
+    let response = this.response;
+
+    trace('Geocoding', address);
+    geocoder.geocode(address)
+    .then(function(res) {
+        trace(res);
+        response.json([['latitude', res[0].latitude], ['longitude', res[0].longitude]]);
+    })
+    .catch(function(err) {
+        error('Error in geocoding', err);
+        showError('Failed to geocode',response);
+    });
+
+    return null;
+};
+
+// reverse geocode and send back a specific detail
+GeoLocationRPC.city = function (latitude, longitude) {
+    reverseGeocode(latitude, longitude, this.response, '.city');
+    return null;
+};
+
+
+// reverse geocode and send back a specific detail
+GeoLocationRPC.country = function (latitude, longitude) {
+    reverseGeocode(latitude, longitude, this.response, '.country');
+    return null;
+};
+
+// reverse geocode and send back a specific detail
+GeoLocationRPC.countryCode = function (latitude, longitude) {
+    reverseGeocode(latitude, longitude, this.response, '.countryCode');
+    return null;
+};
+
+// find places near a coordinate (20 reults max)
+GeoLocationRPC.nearbySearch = function (latitude, longitude, keyword, radius) {
+    let response = this.response;
+    radius = radius || 50000; // default to 50KM
+
+    let requestOptions = {
+        method: 'get',
+        uri: 'https://maps.googleapis.com/maps/api/place/nearbysearch/json',
+        qs: {
+            location: latitude + ',' + longitude,
+            radius: radius,
+            // rankby: 'distance',
+            key: GEOCODER_API
+        },
+        json: true
+    };
+
+    if (keyword) {
+        requestOptions.qs.keyword = keyword;
+    }
+
+    trace('Doing a nearby search', requestOptions);
+    rp(requestOptions).then(res=>{
```

Comment:
I would return this rather than `null` - it should be more clear what is happening